.github/workflows/ci.yaml

```diff
@@ -35,14 +35,19 @@ jobs:
       - name: Install dependencies
         run: poetry install --no-interaction
 
-      - name: Lint (flake8)
-        run: poetry run pre-commit run flake8 --all-files --show-diff-on-failure
-
-      - name: Order of imports (isort)
-        run: poetry run pre-commit run isort --all-files --show-diff-on-failure
-
-      - name: Formatting (black)
-        run: poetry run pre-commit run black --all-files --show-diff-on-failure
+      - name: Lint (ruff)
+        shell: bash
+        run: |
+          if [[ "$RUNNER_OS" == "Linux" && "${{ matrix.python-version }}" == "3.10" ]]
+          then
+            echo "::add-matcher::.github/ruff-matcher.json"
+            poetry run ruff check . --config pyproject.toml --output-format text --no-fix
+          else
+            poetry run pre-commit run ruff --all-files --show-diff-on-failure
+          fi;
+
+      - name: Formatting (ruff)
+        run: poetry run pre-commit run ruff-format --all-files --show-diff-on-failure
```

Comment:
Maybe you can also add some docs telling how people can configure ruff as their default formatter?

Reply:
- https://plugins.jetbrains.com/plugin/20574-ruff
- https://github.com/astral-sh/ruff-vscode

Reply:
This can be the first draft of contribution.md

Reply:
Sure